### PR TITLE
Remove draft petition after case submission

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-644

### Change description ###

- Includes functionality to remove draft petition after case submission.
- Includes unit and integration tests.
- Tested end to end on local docker.
- Includes fix when payment is not present else about to submit will fail to complete.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java [hunk 1]
```diff
@@ -138,7 +139,8 @@ private void updateRespondentDigitalDetails(CaseData caseData) {
     public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
                                                CaseDetails<CaseData, State> beforeDetails) {
         final CaseData caseData = details.getData();
-        final int feesPaid = caseData.getPayments().stream()
+        final List<ListValue<Payment>> payments = caseData.getPayments() == null ? emptyList() : caseData.getPayments();
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:142]: Is this change covered by a unit test?

File: src/test/java/uk/gov/hmcts/divorce/solicitor/service/DraftPetitionRemovalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            userId,
+            documentUuid,
+            true
+        );
+
+        verifyNoMoreInteractions(idamService, authTokenGenerator, documentManagementClient);
+    }
+
+    @Test
+    public void shouldThrow403ForbiddenWhenServiceIsNotWhitelistedInDocManagement() {
```
Comment (mapingo) [inline src/test/java/uk/gov/hmcts/divorce/solicitor/service/DraftPetitionRemovalServiceTest.java:112]: Would it be better to test the 403 and 401 exceptions at the integration level?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:142]: it was existing code which was failing while testing on docker. I think there is a test when payments is present let me add a test when there are no payments

**Review by mapingo — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:142]: this is done now

**Pull request merged**